src/logic/tokens/hooks/useToken.ts

```diff
@@ -0,0 +1,97 @@
+import { useDispatch, useSelector } from 'react-redux'
+import { useEffect, useState } from 'react'
+import ERC20Detailed from '@openzeppelin/contracts/build/contracts/ERC20Detailed.json'
+import { fetchToken } from 'src/logic/tokens/api'
+import { tokensSelector } from 'src/logic/tokens/store/selectors'
+import { makeToken, Token } from 'src/logic/tokens/store/model/token'
+import generateBatchRequests from 'src/logic/contracts/generateBatchRequests'
+import { ETH_ADDRESS } from 'src/logic/tokens/utils/tokenHelpers'
+import etherLogo from 'src/assets/icons/icon_etherTokens.svg'
+import { addToken } from 'src/logic/tokens/store/actions/addToken'
+import { nftAssetsListSelector } from 'src/logic/collectibles/store/selectors'
+import { NFTAsset } from 'src/logic/collectibles/sources/OpenSea'
+
+const getTokenInfoFromBlockchain = (tokenAddress: string): string[] =>
+  generateBatchRequests({
+    abi: ERC20Detailed.abi,
+    address: tokenAddress,
+    methods: ['decimals', 'name', 'symbol'],
+  })
+
+export const useToken = (tokenAddress: string, isCustomToken = false): Token | NFTAsset => {
+  const [token, setToken] = useState<Token | NFTAsset | null>(null)
```

Comment:
Is it possible to unify this `Token` and `TokenDTO`? perhaps making some props on `TokenDTO` optional

Reply:
As I told you above DTO stays for Data Transfer Object and we use that type to represent the data we receive from the backend. TokenProps type is a type for describing props we have in Immutable.js record. Token is a type to represent immutable.js record of a token

Reply:
@nicosampler, we used to name those DTO as `<something>ServiceModel`... that being said, I'm ok with the `DTO`, even it's easier to remember.

Reply:
In that case, it's better to follow the already existing convention. Sorry, I missed it